src/KM_Controls.pas

```diff
@@ -6770,17 +6844,19 @@ procedure TKMMasterControl.SetCtrlFocus(aCtrl: TKMControl);
   begin
     if fCtrlFocus <> nil then
     begin
-      if  Assigned(fCtrlFocus.fOnFocus) then
-        fCtrlFocus.fOnFocus(False);
+      fCtrlFocus.FocusChanged(False);
+      if  Assigned(fCtrlFocus.fOnFocusChange) then
+        fCtrlFocus.fOnFocusChange(False);
       // Reset Parent Panel FocusedControlIndex only for different parents
       if (aCtrl = nil) or (aCtrl.Parent <> fCtrlFocus.Parent) then
         fCtrlFocus.Parent.ResetFocusedControlIndex;
     end;
 
     if aCtrl <> nil then
     begin
-      if Assigned(aCtrl.fOnFocus) then
-        aCtrl.fOnFocus(True);
+      aCtrl.FocusChanged(True);
+      if Assigned(aCtrl.fOnFocusChange) then
+        aCtrl.fOnFocusChange(True);

```

Comment:
as we added FocusChanged procedure and renamed OnFocus event into OnFocusChange - update its invokations here